reana_workflow_engine_yadage/externalbackend.py

```diff
@@ -38,41 +37,6 @@ def get_commands(job):
     return _prettified_cmd, _wrapped_cmd
 
 
-class ExternalProxy(PacktivityProxyBase):
-    """REANA yadage external proxy."""
-
-    def __init__(self, job_id, spec, pars, state):
-        """Initialize yadage external proxy."""
-        super().__init__()
-        self.job_id = job_id
-        self.spec = spec
-        self.pars = pars
-        self.state = state
-
-    def proxyname(self):
-        """Return the proxy name."""
-        return 'ReanaExternalProxy'
-
-    def details(self):
-        """Retrieve the proxy details."""
-        return {
-            'job_id': self.job_id,
-            'spec': self.spec,
-            'pars': self.pars.json(),
-            'state': self.state.json(),
-        }
-
-    @classmethod
-    def fromJSON(cls, data):
-        """Retrieve proxy details from JSON."""
-        return cls(
-            data['proxydetails']['job_id'],
-            data['proxydetails']['spec'],
-            data['proxydetails']['pars'],
-            data['proxydetails']['state']
-        )
-
-
 class ExternalBackend(object):
```

Comment:
Let's use the [`ExternalAsyncMixin`](https://github.com/yadage/packtivity/blob/ac7f544a9f0e0e9cdafa7986c30e38e4ea4275bc/packtivity/asyncbackends.py#L85) which will help us to decouple the job spec creation from the job submission in the `submit` method.